ts/src/facilityCore.ts

```diff
@@ -35,7 +35,7 @@ export interface IHttpClientOptions {
 export namespace HttpClientUtility {
 	/** The fetch function. */
 	export interface IFetch {
-		(uri: string, request: IFetchRequest): Promise<IFetchResponse>;
+		(uri: string, request: IFetchRequest, context?: any): Promise<IFetchResponse>;
```

Comment:
You could use `unknown` instead of `any`, which would force a type check/cast where it's used.

It's nice that `IFetchRequest` is compatible with [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/fetch)'s `init`. Did you consider adding a `signal` of type `any` or `unknown` to that interface instead of separate parameter?

Reply:
> You could use `unknown` instead of `any`, which would force a type check/cast where it's used.

That sounds nice.

Reply:
> Did you consider adding a `signal` of type `any` or `unknown` to that interface instead of separate parameter?

I'm not intuitively excited about the idea, but I'm willing to be convinced. I like the context being ignored unless the caller explicitly does something with it. I wouldn't want people to get the idea that it "should" be an `AbortSignal`.

Reply:
```js
fetch(uri, { ...request, signal: context as AbortSignal? })
```
It seems fine.

`node-fetch` doesn't have `signal` in its `RequestInit` type anyway, at least in the version I happen to have in the project I'm currently working on.